pkg/autoupdate/tuf_autoupdate.go

```diff
@@ -0,0 +1,211 @@
+package autoupdate
+
+// This new autoupdater points to our new TUF infrastructure, and will eventually supersede
+// the legacy `Updater` in autoupdate.go that points to Notary.
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/version"
+	legacytuf "github.com/kolide/updater/tuf"
+	client "github.com/theupdateframework/go-tuf/client"
+	filejsonstore "github.com/theupdateframework/go-tuf/client/filejsonstore"
+)
+
+//go:embed assets/tuf-dev/root.json
+var rootJson []byte
+
+const (
+	DefaultTufServer = "https://tuf-devel.kolide.com"
+)
+
+type TufAutoupdater struct {
+	metadataClient  *client.Client
+	mirrorClient    *http.Client
+	mirrorUrl       string
+	binary          string
+	operatingSystem string
+	channel         UpdateChannel
+	checkInterval   time.Duration
+	errorCounter    []int64
+	lock            sync.RWMutex
+	interrupt       chan struct{}
+	logger          log.Logger
+}
+
+type TufAutoupdaterOption func(*TufAutoupdater)
+
+func WithTufLogger(logger log.Logger) TufAutoupdaterOption {
+	return func(ta *TufAutoupdater) {
+		ta.logger = log.With(logger, "component", "tuf_autoupdater")
+	}
+}
+
+func WithChannel(channel UpdateChannel) TufAutoupdaterOption {
+	return func(ta *TufAutoupdater) {
+		ta.channel = channel
+	}
+}
```

Comment:
Right now we've got 1 and 2, and I imagine 3 would (eventually) be handled by a new consumer as a part of phase 5-ish. I can plan on coming back to runtime in part 5?

Reply:
Huh, I guess it could be a subscriber... Though I think it might be cleaner to just see where the runtime interfaces get us. Wither way, this is fine for now